can/io/trc.py

```diff
@@ -308,9 +322,15 @@ def __init__(
         self.header_written = False
         self.msgnr = 0
         self.first_timestamp = None
-        self.file_version = TRCFileVersion.V2_1
-        self._msg_fmt_string = self.FORMAT_MESSAGE_V1_0
-        self._format_message = self._format_message_init
+        self._setup_file_version(file_version)
+
+    def _setup_file_version(self, file_version: Union[int, TRCFileVersion]) -> None:
+        try:
+            self.file_version = TRCFileVersion(file_version)
```

Comment:
instance attributes should be set in `__init__`. You could have version and format string as return values of a static method or just remove this method and put the try/except block into `__init__`

Reply:
ok. The branch where i picked this from handled it with a tuple returned. I think this does not improve readability.

As this method is invoked from the __init__ function it is always ensured that the attributes are initialized. Can you tell my why it should be added explicit in __init__?

Reply:
I googled a bit for this and it seems like the only requirement is that all attributes are already initialized after the __init__ function was executed. It is not enforced that the assignment is inside of the __init__ function. So for me it looks like the approach here is correct.

I prefer to try to make a function implementation fit on one or two screen page. With this implementation we are quite prepared for more additions and changes not to blow up the function.

If we return a tuple we would achieve the same that is correct, but i think it is better readable like this.

So please take another look and tell me your thoughts.

Reply:
My thoughts are still the same. Instance attributes belong inside `__init__`. Here is the output of pyrefly for example:
```
ERROR /home/user/repositories/python-can/can/io/trc.py:317:13-34: Object of class `TextIO` has no attribute `reconfigure` [missing-attribute]
ERROR /home/user/repositories/python-can/can/io/trc.py:329:18-30: Attribute `file_version` is implicitly defined by assignment in method `_setup_file_version`, which is not a constructor [implicitly-defined-attribute]
ERROR /home/user/repositories/python-can/can/io/trc.py:330:18-33: Attribute `_msg_fmt_string` is implicitly defined by assignment in method `_setup_file_version`, which is not a constructor [implicitly-defined-attribute]
```

So, it would be nice if you fix it, so i can merge.